packages/batch-submitter/src/batch-submitter/state-batch-submitter.ts

```diff
@@ -51,6 +59,49 @@ export class StateBatchSubmitter extends BatchSubmitter {
     return
   }
 
+  public async _getBatchStartAndEnd(): Promise<Range> {
+    const startBlock: number = parseInt(
+      await this.chainContract.getTotalElements(),
+      16
+    )
+    // We will submit state roots for txs which have been in the tx chain for a while.
+    const callBlockNumber: number =
+      (await this.signer.provider.getBlockNumber()) - this.finalityConfirmations
+    const getTotalElementsTx: string = this.ctcContract.interface.encodeFunctionData(
+      'getTotalElements',
+      []
+    )
+    const totalElements: number = parseInt(
+      await this.signer.provider.call(
+        {
+          to: this.ctcContract.address,
+          data: getTotalElementsTx,
+        },
+        callBlockNumber
+      ),
+      16
+    )
```

Comment:
Can't you do `this.ctcContract.getTotalElements()`?

Reply:
Ah great question @gakonst  -- so I want to call `getTotalElements()` but I want to do it based on a _finalized_ Ethereum block. To do this I am running the `call` but passing in the old block number (specifically `callBlockNumber`) which as you can see above is `(await this.signer.provider.getBlockNumber()) - this.finalityConfirmations`